### PR TITLE
fix(new): command to npm link to `angular-cli`

### PR DESCRIPTION
**Pull request by monojack**
add `--link-cli` command to link newly generated project to the global `angular-cli`
package. reduces `ng new ...` command time with about 50%

### CODE REVIEW

**Comment by monojack** [comment]
I signed it!

**Comment by monojack** [comment]
Yea, it was me and I approve with this commit

**Comment by monojack** [comment]
I'm the same guy. I approve

**Comment by monodnb** [comment]
It's me, I'm the same person, I approve with the commits

File: addon/ng2/tasks/link-cli.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  run: function () {
+    var chalk = require('chalk');
+    var ui = this.ui;
+
+    return exec('npm link angular-cli')
+      .then(function () {
+        ui.writeLine(chalk.green('Successfully linked to angular-cli.'));
+      })
+      .catch(function (/*error*/) {
+        ui.writeLine(chalk.red(
```
Comment (hansl) [inline addon/ng2/tasks/link-cli.js]: It all fits in one line, like the `chalk.green` above.

File: addon/ng2/tasks/link-cli.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/* jshint node: true */
+'use strict';
+
+var Promise = require('ember-cli/lib/ext/promise');
+var Task = require('ember-cli/lib/models/task');
+var exec = Promise.denodeify(require('child_process').exec);
+
+module.exports = Task.extend({
+  run: function () {
+    var chalk = require('chalk');
```
Comment (hansl) [inline addon/ng2/tasks/link-cli.js]: Place this require at the top, please.

File: addon/ng2/tasks/link-cli.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/* jshint node: true */
```
Comment (hansl) [inline addon/ng2/tasks/link-cli.js]: Tasks can be TypeScript.

File: addon/ng2/tasks/link-cli.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+'use strict';
```
Comment (hansl) [inline addon/ng2/tasks/link-cli.js]: TypeScript doesn't require use strict, as it is strict by default.

File: addon/ng2/tasks/link-cli.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+var Promise = require('ember-cli/lib/ext/promise');
```
Comment (hansl) [inline addon/ng2/tasks/link-cli.js]: `import * as Promise from '...'`

File: addon/ng2/commands/init.js [hunk 6]
```diff
@@ -57,6 +59,14 @@ module.exports = Command.extend({
       });
     }
 
+    if (commandOptions.linkCli) {
```
Comment (hansl) [inline addon/ng2/commands/init.js]: You should create that closer to where you use it. It's also weird (and will be invalid) to use `var linkCli` inside a scope block.

Also, use `let` for local variables to catch these.

**Comment by hansl** [comment]
Last comment:

In the `Can change current working directory to 'test-project'` test could you use this option instead of what we're currently doing (which is the same thing but manually).

**Comment by hansl** [comment]
Can you rebase? I'll merge this when it's ready (or rebase myself later today).

**Comment by monojack** [comment]
I was away, sorry.
It's done, Rebased, ready for merge. Cheers!

**Comment by aecz** [comment]
Looks like the npm link is done after the npm install which defeats the purpose as it does not reduce install time nor size of node_modules. @filipesilva @monojack @monodnb Please review.

As a workaround, i use:
```
ng new --link-cli --skip-npm myproject
cd myproject
npm install
```
Woud love to have it in one command. Thanks.

**Comment by kylecordes** [comment]
With from @aecz 's process, link-cli is a large improvement in project setup time. Might it be worthwhile to make that process the default you get with "ng new"? In other words, make the fast path default, and the slow (but more correct, from a version-locking point of view) optional? It seems this really improves the CLI user experience. (Currently running ng-new is pretty fast on my Macbook, not so fast on most of Windows machines.)

**Comment by monojack** [comment]
Made a new pull request to address this
#2250

**Pull request merged**